pkg/kubectl/cmd/annotation.go

```diff
@@ -0,0 +1,288 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/spf13/cobra"
+)
+
+// AnnotateOptions have the data required to perform the annotate operation
+type AnnotateOptions struct {
+	out               io.Writer
+	resources         []string
+	annotationArgs    []string
+	newAnnotations    map[string]string
+	removeAnnotations []string
+	namespace         string
```

Comment:
Actually, it might be easier to have composers create this object with `*resource.Builder` that is "bare" - only create by `New` or even the mapper, typer, client mapper.  It looks like the impl is pretty dependent on exactly how the `resource.Builder` is made.

You can see how we've used here: https://github.com/openshift/origin/blob/master/pkg/config/cmd/cmd.go#L14.